.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,119 @@
+name: Create and deploy ext2 image to pages
+
+# Controls when the workflow will run
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+  # Lets you supply arguments when you run a github action manually.
+    inputs:
+      dockerfile_context_path:
+        description: 'Path or url to the Dockerfile its context.'
+        required: true
+        default: 'dockerfiles'
+
+      dockerfile_path:
+        description: 'Path to the Dockerfile.'
+        required: true
+        default: 'dockerfiles/Dockerfile'
+  
+      image_size:
+        description: 'Size of the base ext2 image. 1G is the max and recommended size. Reference the dd manpage for multiplicative suffixes'
+        required: true
+        default: '1G'
+
+      image_name:
+        description: 'Name of final image. Example final url: https://www.github.io/<owner>/<repositoryName>/<image_name>'
+        required: true
+        default: 'base_image.ext2'
+
+      deploy_to_pages:
+        description: 'Set to true to deploy to pages.'
+        required: true
+        default: 'true'
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest # Image to run the worker on.
+    
+    # Our variables
+    env:
+      ext2_img: "ext2-webvm-base-image"
+      dockerfile_context_path: '${{ github.event.inputs.dockerfile_context_path }}'
+      dockerfile_path: '${{ github.event.inputs.dockerfile_path }}'
+
+      tag: "ext2_img"
+      IMAGE_SIZE: '${{ github.event.inputs.image_size }}'
+      IMAGE_NAME: '${{ github.event.inputs.image_name }}'
+      DEST: /assets/images/tmp/ # Path to directory where we host the final image from.
+  
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so our job can access it
+      - uses: actions/checkout@v3
+      # We make the directory to host our image from.
+      - run: sudo mkdir -p /assets/images/tmp/
+      # We build the i386 Dockerfile image.
+      - run: docker build $dockerfile_context_path --file $dockerfile_path --tag $tag --platform=i386 
+      # We run our image with tail -f /dev/null to keep it running.
+      - run: docker run --rm -d $tag tail -f /dev/null
+      # We create our ext2 filesystem.
+      # We allocate space for the ext2 image
+      - run: sudo dd if=/dev/zero of=${DEST}${IMAGE_NAME} bs=1 count=0 seek=$IMAGE_SIZE
```

Comment:
This is likely inefficient as you are writing 1 byte at the time. Can we do better?